nuxeo-features/nuxeo-automation/nuxeo-automation-features/src/main/java/org/nuxeo/ecm/automation/core/operations/users/SuggestUserEntries.java

```diff
@@ -269,8 +275,20 @@ public Blob run() throws IOException {
                 user = userManager.getUserModel(userId);
                 UserAdapter userAdapter = user.getAdapter(UserAdapter.class);
                 List<String> groups = userAdapter.getGroups();
-                if (groups != null && groups.contains(groupRestriction)) {
-                    result.add(obj);
+                List<String> restrictedGroups = new ArrayList<>();
+                restrictedGroups.add(groupRestriction);
+
+                if (groups != null) {
```

Comment:
I think we can also test if `goups` is not empty before computing the descendant groups, to avoid computing them uselessly for a user that isn't in any group.